library/core/src/intrinsics.rs

```diff
@@ -841,11 +841,21 @@ extern "rust-intrinsic" {
     /// Both types must have the same size. Neither the original, nor the result,
     /// may be an [invalid value](../../nomicon/what-unsafe-does.html).
     ///
+    /// When transmuting containers, make sure to not violate the container invariants too.
+    /// Some containers might rely on the size of the type, alignment, or even the `TypeId`,
+    /// in which case transmuting wouldn't be possible without violating the container invariants.
```

Comment:
Oh, you moved it all the way up here, that was a misunderstanding... I think this is lass important than some of the comments that follow, this is is too prominent of a position IMO.

What I meant was to put this together with "Turning a `Vec<&T>` into a `Vec<Option<&T>>`". And I meant the entire paragraph; now the "To transmute the inner type of the contents of a container," still exists down there and duplicates some of the text, which seems redundant.